crates/symbolicator/src/cache.rs

```diff
@@ -117,6 +121,56 @@ impl CacheStatus {
     }
 }
 
+/// A remote cache that can be shared between symbolicator instances.
+#[derive(Debug, Clone)]
+pub struct SharedCache {
+    config: SourceConfig,
+    downloader: Arc<DownloadService>,
+}
+
+impl SharedCache {
```

Comment:
very high level mulling; one of the things i found really difficult to keep straight in my brain when i was working on caches was that there was an implicit categorization of caches: the "data" caches and the derived caches. 

if i understand correctly, this basically means that after this PR now have the following:

### data cache
- object cache
- bitcode?
- this new shared cache

### derived cache
- symcache
- cficache

while there is some code shared between every item listed above irrespective of their "category" by virtue of them all being caches, i feel that their category also has a meaningful effect on how and where they are used. 

i kind of wish the code itself had a way to explicitly declare this categorization, and the addition of a new type of cache i feel makes it more important to do so. i don't think it really fits within the scope of this PR, or possibly even this project; just hoping that this sparks some discussion. food for thought, let's say.

Reply:
the distinction between data and derived cache is only in the way they are handling object expiry. the shared cache is completely agnostic about that and does not fit into any of those categories.

but on the topic of expiration: the idea was that we only cache "positive" downloads/computations, and have a TTL that is equal to our normal data retention periods. That way, we won’t have to worry about expiration at all.

Currently the different forms of caches have different settings there, and also different settings for positive vs negative (not found, error, etc). we focus only on one case here.